Feature#120 fegin client

gkwjdtn975 opened the pull request:
<!-- .github/pull_request_template.md -->

## 📝 PR 개요

회원가입시 쿠폰 발급을 위한 FeignCoupon 구현
FeignCoupon - userServiceImpl 적용 

## 🔗 관련 이슈
#120 

## ✅ 체크리스트

- [ ] 코드가 정상적으로 동작합니다.
- [x] 새로운/수정된 기능에 대한 테스트를 추가했습니다.
- [ ] 문서(README 등)를 최신 상태로 반영했습니다.
- [x] 코드 스타일 가이드 및 커밋 메시지 규칙을 준수했습니다.

## 🖼️ 변경 전/후 스크린샷 (UI 변경 시, 선택)

- 변경된 UI가 있다면 스크린샷을 첨부해 주세요.

## 💬 기타 참고 사항

- 리뷰어가 참고하면 좋을 추가 정보나 특이사항을 적어 주세요.

gkwjdtn975 committed:
fix(payment) FeignClient - UserServiceImpl 연결 구현

gkwjdtn975 committed:
feat: 쿠폰 서비스 연동을 위한 CuponClient 인터페이스 추가

gkwjdtn975 committed:
feat: UserServiceTest.java -> CouponClient를 Mock 객체로 추가하고

gkwjdtn975 committed:
feat(config-repo): CouponClient.java- user-api에서 할 필요 없어서 삭제

gkwjdtn975 committed:
feat(config-repo): CouponClient.java- user-api에서 할 필요 없어서 삭제

gkwjdtn975 committed:
feat(payment): LoadBalanced 사용해 user-api서비스 이름으로 호출(회원 정보 가져오기 위해 작성)

gkwjdtn975 committed:
feat : 로드 밸런싱을 위한 쿠폰 API 호출

gkwjdtn975 committed:
feat: 쿠폰 발급을 하기전 유저 정보를 받기위한 couponclient 구현

gkwjdtn975 committed:
feat: 쿠폰 발급을 하기전 유저 정보를 받기위한 couponclient 구현

gkwjdtn975 committed:
Merge branch 'develop' into feature#120-Fegin-Client

gkwjdtn975 committed:
feat:

mni-js commented on src/main/java/com/nhnacademy/bookstoreuserapi/BookstoreUserApiApplication.java:
```diff
@@ -16,4 +19,9 @@ public static void main(String[] args) {
 		SpringApplication.run(BookstoreUserApiApplication.class, args);
 	}
 
+	@Bean
```
사용하는 곳이 없다면 삭제해도 될 것 같습니다.

gkwjdtn975 replied:
확인했습니다

JYCuzz commented on src/main/java/com/nhnacademy/bookstoreuserapi/client/CouponClient.java:
```diff
@@ -0,0 +1,26 @@
+import org.springframework.web.bind.annotation.PostMapping;
+import lombok.extern.slf4j.Slf4j;
+
+@FeignClient(name = "gateway-service", fallback = CouponClient.CouponClientFallback.class)
```
api 끼리 통신할땐 gateway 를 거칠 필요가 없을 것 같은데 다이렉트로 통신해보시는건 어떠실까요?

gkwjdtn975 replied:
정보 감사합니다 수정했습니다.

gkwjdtn975 committed:
feat: 결합도를 낮추기 위해 제거

gkwjdtn975 committed:
feat: RabbitMQ 의존성 추가

gkwjdtn975 committed:
feat: RabbitMQ properties 정보 추가 작성

gkwjdtn975 committed:
feat: RabbitMQ properties 추가했으므로 restTemplate 삭제

gkwjdtn975 committed:
feat: FeignClient 의존성 추가

gkwjdtn975 committed:
feat: 회원가입시 쿠폰 발급 (이벤트)처리를 위한 RqbbitTemplate추가

gkwjdtn975 committed:
feat: userId값을 받아오기 위해 userNo 필드 추가

gkwjdtn975 committed:
feat: 사용자 아이디로 userNo 조회 API 추가
-userController에 GET /id/{userId}엔트포인트 구현

gkwjdtn975 committed:
feat: userId 값을 받아서, 해당 사용자의 아이디와 내부 식별자인 userNo를 담은 ResponseUserId Dto 객체 반환을 위한 추가  (userID 문자열 대신 userno 고유넘버로 조회)

gkwjdtn975 committed:
feat: 클라이언트가 요청한 userId 에 대응하는 사용자 정보 조회 없으면 예외처리 추가

gkwjdtn975 committed:
feat: 유저가 가입됐을 때 이벤트처리를 하기위한 추가
- publisher(유저): 회원가입 시 rabbitTemplate.converAndSend()로 메시지 가능

gkwjdtn975 committed:
feat: 사용자 등록 이벤트 `UserRegisteredEvent` 추가

- `UserRegisteredEvent` 클래스 생성
- `userNo` 필드로 가입된 사용자 번호 전달
- 직렬화를 위해 `Serializable` 구현

gkwjdtn975 committed:
feat: UserIdByUserId 불필요 삭제

gkwjdtn975 committed:
feat: UserIdByUserId 불필요 삭제

gkwjdtn975 committed:
feat: UserIdByUserId 불필요 삭제

gkwjdtn975 committed:
feat: ResponseUser에서 받아오기로 결정해서 삭제

gkwjdtn975 committed:
Merge branch 'develop' into feature#120-Fegin-Client

# Conflicts:
#	src/main/java/com/nhnacademy/bookstoreuserapi/user/controller/UserController.java

gkwjdtn975 committed:
feat: test코드에 RabbitTemplate가 들어가면 안되기 때문에 configServe로 옮김

gkwjdtn975 committed:
feat: rabbitTemplate Mock 추가

gkwjdtn975 committed:
feat: config로 이동

JYCuzz commented on src/main/resources/application-dev.properties:
```diff
@@ -2,4 +2,18 @@ spring.datasource.url=jdbc:mysql://10.116.64.14:13306/project_be10_team3_dev?use
+spring.rabbitmq.port=5672
+spring.rabbitmq.username=admin
+spring.rabbitmq.password=nhnacademy123!
+spring.rabbitmq.virtual-host=/
```
rabbitmq 도 민감정보는 config.repo 에 보관하고 config.server 를 통해 가져오는게 좋을 것 같아요

gkwjdtn975 replied:
감사합니다

JYCuzz reviewed (approved):
나중에 민감정보 변경하시면 될것같습니다
